.github/workflows/release.yml

```diff
@@ -0,0 +1,357 @@
+                const draft_found = !(draft === undefined)
+                
+                let release
+                if (draft_found){
+                console.log("Draft release already exists. Deleting current draft release and recreating it")
+                release = (await github.rest.repos.deleteRelease({
+                    owner: context.payload.repository.owner.login,
+                    repo: context.payload.repository.name,
+                    release_id: draft.id,
+                }))
+                }
+                
+                const release_notes = (await github.rest.repos.generateReleaseNotes({
+                    owner: context.payload.repository.owner.login,
+                    repo: context.payload.repository.name,
+                    tag_name: "v"+version,
+                    previous_tag_name: latest_release,
+                    target_commitish: ref,
+                }))
+                
+                const footer = `
+            ## Resources
+            - Documentation -- https://github.com/nginx/agent#readme
+                `
+                
+                release = (await github.rest.repos.createRelease({
+                    owner: context.payload.repository.owner.login,
+                    repo: context.payload.repository.name,
+                    tag_name: "v"+version,
+                    target_commitish: ref,
+                    name: "v"+version,
+                    body: release_notes.data.body + footer,
+                    draft: true,
+                }))
+                
+                console.log(`Release created: ${release.data.html_url}`)
+                console.log(`Release ID: ${release.data.id}`)
+                console.log(`Release notes: ${release_notes.data.body}`)
+                console.log(`Release Upload URL: ${release.data.upload_url}`)
+                
+                return {
+                    version: version,
+                    release_id: release.data.id,
+                    release_upload_url: release.data.upload_url,
+                }
+            }
+
+      - name: Set Environment Variables
+        id: vars
+        run: |
+          echo "RELEASE_ID=$(echo '${{steps.release.outputs.result}}' | jq -r '.release_id')" >> $GITHUB_OUTPUT
+          cat $GITHUB_OUTPUT
+
+  tag-release:
+    name: Tag Release
+    runs-on: ubuntu-22.04
+    needs: [vars,release-draft]
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+        with:
+          ref: ${{ inputs.releaseBranch }}
+
+      - name: Tag release
+        run: |
+          git config --global user.name 'github-actions'
+          git config --global user.email '41898282+github-actions[bot]@users.noreply.github.com'
+
+          git tag -a "v${{ inputs.packageVersion }}" -m "CI Autogenerated"
+          git tag -a "sdk/v${{ inputs.packageVersion }}" -m "CI Autogenerated"
```

Comment:
For v3 we actually dont have a sdk anymore so we won't need the `sdk/v...` tag.
The sdk tag is only required for v2 releases.

Reply:
Thanks @dhurley I'll remove that tag